### ISSUE TITLE
Improve hyperthreading performance on KNL

### ISSUE THREAD

**Issue by e-kayrakli** [chapel-lang/chapel#6389]
PRKs show suboptimal performance with hyperthreading most of the time. Currently Nstream achieves following bandwidths on master:

| Threads |		BW(GB/s)|
|---|---|
|64|		        368.88|
|128|		        233.02|
|192	|	        304.45|
|256	|	        313.07|

Similar drop is also apparent in other PRKs. Especially 2 thread/core seems to be a local minimum in most cases.

For reference, OpenMP plateaus around 410 GB/s with >=64 threads.

**Comment by ronawho** [comment]
All of the PRK studies were done with quad clustering mode, right? What about HBM, was this with HBM only, or HBM as cache?

**Comment by e-kayrakli** [comment]
Yes, quadrant. Memory mode is cache. The vectors are 43M doubles.
